From fe844fe620892bd247c502956445d87d8df73af0 Mon Sep 17 00:00:00 2001
From: mphys2 <mphys2@psrgpu2.ast.man.ac.uk>
Date: Sat, 25 Feb 2017 11:38:59 +0000
Subject: [PATCH] fixed classify cands to deal with commented lines correctly

---
 src/classify_cands.py |    9 ++++++---
 1 files changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/classify_cands.py b/src/classify_cands.py
index 4f498b0..7bf9645 100644
--- a/src/classify_cands.py
+++ b/src/classify_cands.py
@@ -53,14 +53,17 @@ class ArffBatch(object):
                     if('%' in line):
                         text = line[0:line.index('%')]
                         comment = line[line.index('%'):]
-                        paths.append(comment)
+                        paths.append(comment[1:])
                     else:
                         text = line
                         paths.append("unknown, no path found")
                     # Split on comma since ARFF data is in CSV format.
-                    components = text.split(",")
+                    components = filter(lambda x: x != "", text.split(","))
 
-                    features = [float(x) for x in components[0:len(components)-1]]
+                    try:
+			features = [float(x) for x in components[0:len(components)-1] if x != ""]
+		    except:
+			import pdb; pdb.set_trace()
                     X.append(features)
                     read += 1
                 self.current_pos = f.tell()
-- 
1.7.1

